templates/actors/parts/actor-spellbook.hbs

```diff
@@ -30,91 +30,93 @@
 
 <ol class="items-list inventory-list">
 {{#each spellbook as |section|}}
-    <li class="items-header spellbook-header flexrow">
-        <div class="item-name flexrow">
-            <h3>{{section.label}}</h3>
-            <div class="spell-slots">
-                {{#if section.usesSlots}}
-                <input type="text" name="system.spells.{{section.prop}}.value" value="{{section.uses}}" placeholder="0"
-                       data-dtype="Number">
-                <span class="sep"> / </span>
-                <span class="spell-max" data-level="{{section.prop}}" data-slots="{{section.slots}}">
-                    {{{section.slots}}}
-                    {{#if section.editable}}
-                    <a class="slot-max-override" data-tooltip="DND5E.SpellProgOverride">
-                        <i class="fas fa-edit"></i>
-                    </a>
+    <div {{#each section.dataset as |v k|}}data-{{k}}="{{v}}"{{/each}}>
```

Comment:
This ends up with us repeating these dataset options here and with the `"+ Add"` button. It might be worth it to remove the repeated data below and add a bit of logic to `_onItemCreate` so it can get the properties from either location if necessary.

```js
  _onItemCreate(event) {
    event.preventDefault();
    const dataset = (event.currentTarget.closest(".spellbook-section") ?? event.currentTarget).dataset;
    const type = dataset.type;

    // skipped lines

    const itemData = {
      name: game.i18n.format("DND5E.ItemNew", {type: game.i18n.localize(`ITEM.Type${type.capitalize()}`)}),
      type: type,
      system: foundry.utils.expandObject({ ...dataset })
    };

    // more skipped lines
  }
```

Probably be worth it to add a class here to make it easier to identify the purpose of this div.
```suggestion
    <div class="spellbook-section" {{#each section.dataset as |v k|}}data-{{k}}="{{v}}"{{/each}}>
```